src/store.js

```diff
@@ -78,6 +80,17 @@ export class Store {
     }
   }
 
+  applyMixins (options) {
+    if (Array.isArray(options.mixins)) {
+      const merge = Vue.config.optionMergeStrategies.computed
```

Comment:
I think it is a bad idea to rely on the merge strategy in Vue core because it can easily be broken if the strategy changes the behavior. Let's write it in the Vuex code.
Also, I suspect if we should use `computed` merge strategy since it overwrites a property if there are the same key in the objects. I feel it would be natural to take lifecycle hook merge strategy.